File: Npc/11000183.cs
using Maple2.Model.Enum;
using Maple2.Script.Npc;

/// <summary>
/// 11000183: Queen Bee
/// </summary>
public class _11000183 : NpcScript {
    protected override int First() {
        return 30;
    }

    // Select 0:
    // $script:0831180407000768$
    // - What izzz it?
    protected override int Select() => 0;

    protected override int Execute(int selection) {
        switch (Id, Index++) {
            case (30, 0):
                // $script:0831180407000771$
                // - Honey izz very preciouz to uzz honeybeezz! We won't let anyone take it away!
                return -1;
        }
        
        return default;
    }

    protected override NpcTalkButton Button() {
        return (Id, Index) switch {
            (30, 0) => NpcTalkButton.Close,
            _ => NpcTalkButton.None,
        };
    }
}

File: Npc/11000183.py
""" 11000183: Queen Bee """
from npc_api import Option, Script
import random


class Main(Script):
    def first(self) -> int:
        return 30

    def select(self) -> int:
        return 0

    def __0(self, pick: int) -> int:
        # $script:0831180407000768$
        # - What izzz it?
        return None # TODO

    def __30(self, pick: int) -> int:
        # $script:0831180407000771$
        # - Honey izz very preciouz to uzz honeybeezz! We won't let anyone take it away!
        return -1

    def button(self) -> Option:
        if (self.state, self.index) == (30, 0):
            return Option.CLOSE
        return Option.NONE
